app/sources/HFTemplateController.m

```diff
@@ -380,6 +380,26 @@ - (BOOL)readMacDate:(NSDate **)value forLabel:(NSString *)label {
     return YES;
 }
 
+- (NSString *)readFatDateWithLabel:(NSString *)label error:(NSString **)error {
+    int16_t val;
+    if (![self readInt16:&val forLabel:nil]) {
+        if (error) {
+            *error = @"Failed to read int16 bytes";
+        }
+        return nil;
+    }
+
+    int day = val & 0x1F;
```

Comment:
From looking at https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system it's unclear if value 0 would be month 1, and same for day. When I used a value of all 0s to test, it showed as 1980-0-0 which doesn't appear valid.

Reply:
I glanced at the FAT spec and it says only 1-12 for day and 1-31 for month. Should we add error handling for values outside of those ranges since they don't make sense?

Reply:
You can add an error handling. But not one that ends the execution of the template like `requires`.

I use HexFiend to analyze floppy images. Some are corrupt. It is therefore important for me to be able to see the data even if it is incorrect such as 1980-13-00.

Reply:
Sounds good. I'll leave it as it is.